plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/TestParquetDecimalScaling.java

```diff
@@ -463,6 +508,13 @@ private static String generateTableName(String testCase, int precision, int scal
         return format("%s_%d_%d_%d", testCase, precision, scale, ThreadLocalRandom.current().nextInt(1, MAX_VALUE));
     }
 
+    private static Object[][] withWriterVersion(Object[][] args)
+    {
+        return Arrays.stream(cartesianProduct(args, WriterVersion.values()))
```

Comment:
`org.apache.parquet.column.ParquetProperties.WriterVersion` is not the only thing which determines the file structure, is it?
In #9611  we had to change a few things more.  Please explain.

Reply:
In the linked PR we were changing native trino writer implementation. Here `createParquetFile` uses Hive's RecordWriter to write the parquet files, so just changing WriterVersion works fine. This is similar to `ParquetTester#assertRoundTrip`